### DISCUSSION TITLE
Haspmota: way to attach custom metadata to haspmota objects

### DISCUSSION THREAD

**Discussion by msperl** [arendst/Tasmota#18676] [General]
I was wondering if there is a way to add some custom metadata to haspmota via some extra json attribute, that can be accessed from berry.

I know that "berry_code" is possible, but that is onetime and I have to keep an internal registry and do some mapping code.

With import global/introspection it would make things easier that having to consult a registry for each object (and having to use run_berry on each button/...)

Also as an unfortunate side-effect: contrary to what the documentation says: berry_run seems to be run before initializing the haspmota object (the object does not show up in global (yet) - so one has to defer processing anyway...)

Idea could be:
```
{ "id": 42, "obj": "btn",...,"meta": { <whatever I like here> } }
```
and a getter/setter method: meta for berry

Having this on the lvgl side would make it even more flexible, as then this can get shared between lvgl and specific haspmota code....

**Comment by s-hadinger** [comment]
Thanks for your suggestion. I propose to do the following:
- add a `meta` attribute with free unparsed content
- defer running `berry_run` until the object is parsed and built, but before the entire page is built

**Reply by msperl** [reply]
well, that looks promising if/when it is implemented.

**Reply by s-hadinger** [reply]
It is now implemented, see #18685

**Reply by msperl** [reply]
thanks - building development image right now!

**Reply by msperl** [reply]
thanks It works (also on the pages.jsonl level)!

Now I just need to change my implementation again to make it more streight-forward ;)
